src/hs_util.c

```diff
@@ -97,18 +98,47 @@ unsigned hs_disk_free_ob(const char *path, unsigned ob_size)
 }
 
 
-
-static FILE* common_termination_err(const char *path, const char *name)
+static char* get_termination_fn(const char *path, const char *name)
 {
   if (!path || !name) return NULL;
   const char *pos = strchr(name, '.');
   if (!pos) return NULL;
+  char *fn = malloc(HS_MAX_PATH);
+  if (!fn) return NULL;
+
+  int ret = snprintf(fn, HS_MAX_PATH, "%s/%.*s/%s%s", path, (int)(pos - name),
+                     name, pos + 1, hs_err_ext);
+  if (ret < 0 || ret > HS_MAX_PATH - 1) {
+    free(fn);
+    fn = NULL;
+  }
+  return fn;
+}
+
+
+bool hs_is_bad_state(const char *path, const char *name, const char *state)
+{
+  bool retry = false;
+  char *fn = get_termination_fn(path, name);
+  if (fn) {
+    char *em = lsb_read_file(fn);
+    const char *pos = strstr(em, "restore_global_data");
+    if (pos) {
+      unlink(fn);
```

Comment:
should this be checked for failure and handled?

Reply:
We could check for failure but we really have no way to specifically handle it.  Currently if it fails, the plugin load will fail so the right thing will happen (you will see two consecutive state restoration failures and the plugin will fail to load).